sdk/containerregistry/container-registry/karma.conf.js

```diff
@@ -58,15 +51,16 @@ module.exports = function (config) {
       "TEST_MODE",
       "CONTAINER_REGISTRY_ENDPOINT",
       "CONTAINER_REGISTRY_ANONYMOUS_ENDPOINT",
-      "CONTAINERREGISTRY_CLIENT_ID",
-      "CONTAINERREGISTRY_CLIENT_SECRET",
-      "CONTAINERREGISTRY_TENANT_ID",
+      "AZURE_CLIENT_ID",
+      "AZURE_CLIENT_SECRET",
+      "AZURE_TENANT_ID",
```

Comment:
When I created the weekly live test pipeline, `AZURE_*_*` env vars didn't work for other clouds because they were set to the Public cloud's client/tenant.  It is likely to be fixed now.  Please trigger the `js - container-registry - tests-weekly` pipeline to make sure it's working. https://dev.azure.com/azure-sdk/internal/_build?definitionId=2886

Reply:
I'll trigger that soon. I changed these back to their usual names so that the `createTestCredential` helper could be used -- it uses the standard names when creating the credential. If the issue still exists, then I can revert to using the `CONTAINERREGISTRY_*` names and switch between `NoOpCredential` and the `DefaultAzureCredential` without the helper.

Reply:
I didn't know this.

Maybe update the test-credential package to take more options?

Reply:
Appears to be failing, I'll add an option to test-credential.

Reply:
We may want to fix the pipeline instead? I thought it is expected that `Azure_*` is set to be the ones from the cloud being tests against. @benbp

Reply:
Not blocking this PR. But I'd love to not special case ACR.

Reply:
The change to add options to the test credential is nice to have anyway IMO regardless of whether we fix the underlying issue here.

Reply:
We have not historically set `AZURE_*` variables and instead set namespaced variables like `CONTAINERREGISTRY_*` in the test infastructure to avoid potential collisions when running test batches. I have an issue to fix this though, as in practice it hasn't been an issue. A current workaround is to set a redirect in the pipeline definition, e.g. [here](https://github.com/Azure/azure-sdk-for-js/blob/5593f6b5c0d50db4e927bb21bb0f3bead0dcf1b1/sdk/servicebus/service-bus/tests.yml#L12-L15).

```
        AZURE_CLIENT_ID: $(CONTAINERREGISTRY_CLIENT_ID)
        AZURE_TENANT_ID: $(CONTAINERREGISTRY_TENANT_ID)
        AZURE_CLIENT_SECRET: $(CONTAINERREGISTRY_CLIENT_SECRET)
        AZURE_SUBSCRIPTION_ID: $(CONTAINERREGISTRY_SUBSCRIPTION_ID)
```